include/verilated_types.h

```diff
@@ -539,9 +539,27 @@ template <class T_Key, class T_Value> class VlAssocArray final {
 
 private:
     // MEMBERS
-    Map m_map;  // State of the assoc array
+    mutable Map m_map;  // State of the assoc array
     T_Value m_defaultValue;  // Default value
 
+    std::vector<std::pair<std::size_t, T_Value>> modify_input;
+    mutable bool modified = false;
+
+    const Map& map() const {
+        if (modified) {
+            for (const auto& input : modify_input) {
+                if (input.second)
+                    m_map.emplace(input.first, input.second);
+                else
+                    m_map.erase(input.first);
```

Comment:
I think this assumes that `m_defaultValue` is zero, which is not always true (at least the API explicitly allows changing it).

But more importantly, this breaks for example `exists` if the index was explicitly inserted with a zero value (same for anything that uses `.find`)